File: Calto-GG/Microsoft/Day 5/Minimum_Deletions_to_Make_Array_Divisible_Hard.py
import math
class Solution:
    def minOperations(self, nums: list[int], numsDivide: list[int]) -> int:
        gcd=0
        for i in numsDivide:
            gcd=int(math.gcd(gcd,i))
        nums.sort()
        for i in range(len(nums)):
            if gcd%nums[i]==0:
                return i
        return -1

File: Mayur/Microsoft/Day 5/Minimum_Deletions_to_Make_Array_Divisible_Hard.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, vector<int>& nD) {
        int g=*max_element(nD.begin(),nD.end());
        sort(nums.begin(),nums.end());
        for(auto x:nD)
        {
            g=__gcd(g,x);
        }
        int ans=-1;
        for(int i=0;i<nums.size();i++)
        {
            if(g%nums[i]==0) 
                {
                    ans=i;
                    break;
                }
        }
        return ans;
    }
};